### DISCUSSION TITLE
Idea: Remove the requirements for (implicit) converters to be public

### DISCUSSION THREAD

**Discussion by AdmiralSnyder** [dotnet/csharplang#2763] [Language Ideas]
it would be nice if implicit converters (or converters in general) could be declared non-public i.e. private, protected or internal.
Since it's possible to convert from ValueTuples, this gives the opportunity to make sourcecode much denser for often-used conversions, but it would be nice to restrict the usage of those converters for library-internal use.

my specific usecase is for parameter-collection classes where i convert refactor a call from
` GetDbObjs(new DbObjCreateArgs<Class>(Applic, Db))`
to
`GetDbObjs((Applic, Db))`

I'd like to be able to restrict the usage of those converters to the scopes i want to.

**Comment by YairHalberstadt** [comment]
Target typed new would allow:

```csharp
GetDbObjs(new(Applic, Db))
```

**Comment by AdmiralSnyder** [comment]
if we only had Target typed new already...

**Comment by YairHalberstadt** [comment]
@AdmiralSnyder

Target typed new is scheduled for 8.1, and will almost certainly be released before this feature.

**Comment by AdmiralSnyder** [comment]
this should be noted in here perhaps:
https://github.com/dotnet/csharplang/blob/72a4daa8532e191858f9f887f4c8d280ec006bb4/proposals/target-typed-new.md

**Comment by YairHalberstadt** [comment]
What should be noted in there?

**Comment by 333fred** [comment]
Small nit: @AdmiralSnyder you mean conversions, not converters.
